File: Java/Problem_2.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Problem_2{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of shoe sizes: ");
        int n = scanner.nextInt();

        List<Integer> shoeSizes = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            System.out.print("Enter shoe size " + (i + 1) + ": ");
            int size = scanner.nextInt();
            shoeSizes.add(size);
        }

        int secondLargestSize = findSecondLargest(shoeSizes);

        if (secondLargestSize != Integer.MIN_VALUE) {
            System.out.println("Ram's chosen shoe size: " + secondLargestSize);
        } else {
            System.out.println("There is no second-largest element.");
        }

        scanner.close();
    }

    public static int findSecondLargest(List<Integer> list) {
        if (list.size() < 2) {
            return Integer.MIN_VALUE;
        }

        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        for (int num : list) {
            if (num > largest) {
                secondLargest = largest;
                largest = num;
            } else if (num > secondLargest && num != largest) {
                secondLargest = num;
            }
        }

        if (secondLargest == Integer.MIN_VALUE) {
            return Integer.MIN_VALUE;
        } else {
            return secondLargest;
        }
    }
}

File: Python/Problem_2.py
def find_second_largest(arr):
    if len(arr) < 2:
        return "There is no second-largest element."

    largest = second_largest = float('-inf')

    for num in arr:
        if num > largest:
            second_largest = largest
            largest = num
        elif num > second_largest and num != largest:
            second_largest = num

    if second_largest == float('-inf'):
        return "There is no second-largest element."
    else:
        return second_largest

shoe_sizes = []
n = int(input("Enter the number of shoe sizes: "))

for i in range(n):
    size = int(input(f"Enter shoe size {i + 1}: "))
    shoe_sizes.append(size)

second_largest_size = find_second_largest(shoe_sizes)
print("Ram's chosen shoe size:", second_largest_size)
